### PR TITLE
Basic integration tests for `sov-modules-impl`

### LINKED ISSUE #35: sov-modules: Decide how to combine a module prefix and a state key.
Each module has a globally unique prefix that has to be combined with a state key before db insertion. This way, the same state key in different modules will end up in separate db slots. 

```
pub struct Prefix {
    storage_name: String,
    module_name: String
    module_path: String,
}
```

How do we combine the state key and the prefix? There are two ways:
1. Hash the prefix and concatenate it with the state key: This is a bad idea because it could destroy db caching (we will access many state variables in the same module).
2. Concatenate all the prefix fields and the state key.

TODO: What about `prefix||state_key` padding?

### PR DESCRIPTION
**Pull request by bkolad**
This PR introduces the following changes:
1. Basic integration tests in `sov-modules-impl`.
2. Logic for `pferix` and `state_key` concatentation.
3. Code stub for `AlignedVec `
4. Implementations for `get/set in StateMap`

close #35

### CODE REVIEW

File: sov-modules/sov-modules-api/src/lib.rs [hunk 1]
```diff
@@ -31,12 +31,19 @@ impl Prefix {
 impl From<Prefix> for sov_state::Prefix {
     fn from(prefix: Prefix) -> Self {
         let mut combined_prefix = Vec::with_capacity(
-            prefix.module_path.len() + prefix.module_name.len() + prefix.storage_name.len(),
+            prefix.module_path.len() + prefix.module_name.len() + prefix.storage_name.len() + 3,
```
Comment (preston-evans98) [inline sov-modules/sov-modules-api/src/lib.rs:34]: nit: magic number. Maybe add a `const DOMAIN_SEPARATOR: [u8] = [47]` and then say "+ DOMAIN_SEPARATOR.len() * 3`?

File: sov-modules/sov-state/src/storage.rs [hunk 2]
```diff
@@ -1,25 +1,70 @@
...
+    pub fn new<K: Encode>(prefix: &Prefix, key: K) -> Self {
+        let mut encoded_key = Vec::default();
+        key.encode(&mut encoded_key);
+
+        let encoded_key = AlignedVec::new(encoded_key);
+
+        let full_key = Vec::<u8>::with_capacity(prefix.len() + encoded_key.len());
+        let mut full_key = AlignedVec::new(full_key);
+        full_key.extend(prefix.as_aligned_vec());
+        full_key.extend(&encoded_key);
```
Comment (preston-evans98) [inline sov-modules/sov-state/src/storage.rs:36]: (non-blocking) It seems like we should be able to do this without encoding the key separately and then copying. Maybe we should add a `size_hint` method to the key type?

File: sov-modules/sov-modules-api/src/lib.rs (hunk 1 above)
Comment (bkolad) [reply sov-modules/sov-modules-api/src/lib.rs]: done

**Review by preston-evans98 — APPROVED** [review]

File: sov-modules/sov-state/src/storage.rs (hunk 2 above)
Comment (bkolad) [reply sov-modules/sov-state/src/storage.rs:36]: Resolved offline, we are skipping it for now.

**Pull request merged**